### PR TITLE
aiva2/CCM-5611 cf management routes

### PR DESCRIPTION
**Pull request by aidenvaines-bjss**
## Description

- Refactors the existing infra code to the restructure AMET workflow keeping much of what existed with some variable name changes and file shuffling.
- renames the component as CDN as thats what it does
- the first origin added is the github page for cms-web, ive had to include a lambda@edge function due to the github base needing to remove the path after the domain / but still have it in the origin-request
- https://nhsdigital.github.io/nhs-notify-web-cms/index.html so becomes https://main.web-gateway.nonprod.nhsnotify.national.nhs.uk/ not https://main.web-gateway.nonprod.nhsnotify.national.nhs.uk/nhs-notify-web-cms as that caused a bunch of 404s as. you can imagine
- 
- Adds CI for deploy on merge

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [x] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: infrastructure/terraform/components/cdn/lambda_permission_cloudfront_lambda_remove_origin_request_path.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# resource "aws_lambda_permission" "cloudfront_lambda_remove_origin_request_path" {
```
Comment (sidnhs) [inline infrastructure/terraform/components/cdn/lambda_permission_cloudfront_lambda_remove_origin_request_path.tf:1]: do we want to remove this?

Comment (aidenvaines-bjss) [reply infrastructure/terraform/components/cdn/lambda_permission_cloudfront_lambda_remove_origin_request_path.tf]: Yea, unlike API Gateway, CF creates the trigger automatically. ive dropped the file. Good spot

File: infrastructure/terraform/etc/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# THIS DIRECTORY SHOULD BE IGNORED IN THE CONTEXT OF THE NHS NOTIFY PROGRAMME AND IS INTENDED FOR INDEPENDENT DEVELOPMENT USE ONLY
```
Comment (RossBugginsNHS) [inline infrastructure/terraform/etc/README.md:1]: Could put a gitignore in this folder to exlude .tfvars?

**Review by sidnhs — DISMISSED** [review]

**Comment by m-houston** [comment]
The links used in the pages all end up with the extra path segment, which I assume is not intended - I presume this is because Jekyll is writing the links as absolute links to the underlying hosting base path.

https://main.web-gateway.nonprod.nhsnotify.national.nhs.uk/nhs-notify-web-cms/features/emails

My guess is that we'd need tell Jekyll not to prefix everything to resolve this, which would break the page if viewed from the `github.io` domain. But maybe we could make it work using a custom domain setup for GHP?

**Review by RossBugginsNHS — APPROVED** [review]

**Pull request merged**